homeassistant/components/modbus/binary_sensor.py

```diff
@@ -79,12 +81,19 @@ def device_class(self) -> Optional[str]:
         """Return the device class of the sensor."""
         return self._device_class
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
     def update(self):
         """Update the state of the sensor."""
-        result = self._hub.read_coils(self._slave, self._coil, 1)
         try:
+            result = self._hub.read_coils(self._slave, self._coil, 1)
+            self._available = True
             self._value = result.bits[0]
-        except AttributeError:
+        except (ConnectionException, AttributeError):
```

Comment:
What raises the `AttributeError` here ? Is it because `result` is None` ? If so I prefer we check for that instead of catching AttributeError. Also, if that is the reason we're catching AttributeError, setting `available` to `True` before we read the result might cause a false positive. Setting available to True should be the last action